src/parachain/system.ts

```diff
@@ -8,14 +8,23 @@ export interface SystemAPI {
     /**
      * @returns The current block number being processed.
      */
-    getCurrentBlockNumber(): Promise<BlockNumber>;
+    getCurrentBlockNumber(): Promise<number>;
+    /**
+     * @returns The current active block number being processed.
+     */
+    getCurrentBlockNumber(): Promise<number>;
 }
 
 export class DefaultSystemAPI implements SystemAPI {
     constructor(private api: ApiPromise) { }
 
-    async getCurrentBlockNumber(): Promise<BlockNumber> {
+    async getCurrentBlockNumber(): Promise<number> {
+        const head = await this.api.rpc.chain.getFinalizedHead();
+        return await (await this.api.query.system.number.at(head)).toNumber();
+    }
+
+    async getCurrentActiveBlockNumber(): Promise<number> {
         const head = await this.api.rpc.chain.getFinalizedHead();
-        return await this.api.query.system.number.at(head);
+        return await (await this.api.query.security.activeBlockCount.at(head)).toNumber();
```

Comment:
The first `await` is not required, but it doesn't break anything

Reply:
uh, that was the auto-complete, I'll fix that